@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -572,36 +599,25 @@ export class TopViewClipPlanesEvaluator extends ElevationBasedClipPlanesEvaluato
 }
 
 /**
- * Evaluates camera clipping planes taking into account ground distance and camera angles.
- *
- * @remarks
- * This evaluator provides support for camera with varying tilt (pitch) angle, the angle
- * between camera __look at__ vector and the ground surface normal.
+ * Evaluates camera clipping planes taking into account ground distance and camera tilt (pitch)
+ * angle (angle between look-at vector and ground surface normal).
  */
 export class TiltViewClipPlanesEvaluator extends TopViewClipPlanesEvaluator {
-    private readonly m_ray = new THREE.Ray();
+    private static readonly NDC_BOTTOM_DIR = new THREE.Vector2(0, -1);
+    private static readonly NDC_TOP_RIGHT_DIR = new THREE.Vector2(1, 1);
+
     /**
-     * Calculate the lengths of frustum planes intersection with the ground plane.
-     *
-     * @remarks
-     * This evaluates distances between eye vector (or eye plane in orthographic projection) and
-     * ground intersections of top and bottom frustum planes.
-     * @note This method assumes the world surface (ground) to be flat and
-     * works only with planar projections.
+     * Calculate distances to top/bottom frustum plane intersections with the ground plane.
      *
-     * @param camera - The [[THREE.Camera]] instance in use,
+     * @param camera - The [[THREE.Camera]] instance in use.
      * @param projection - The geo-projection used to convert geographic to world coordinates.
      */
     private getFrustumGroundIntersectionDist(
```

Comment:
I wonder if it makes sense to put this into some sort of `PlanarProj` namespace, like you did with the `SphericalProj` namespace.

Reply:
I thought the same, but since this method uses class members (maxElevation, minElevation), I decided to leave it here.

Reply:
You pass in already `minR: number` & `maxR: number` to the `getFarDistanceFromElevatedHorizon` function in the `SphericalProj` namespace, so passing in elevation isn't too bad, but as you like..